### PR TITLE
refactor: format header files with uncrustify

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

**Comment by justinmk** [comment]
The long lines in `src/nvim/tui/terminfo_defs.h` should not be touched. It's just data. No need for it to be hardwrapped. Can it be marked as `// uncrustify:ignore` or whatever

**Comment by dundargoc** [comment]
Fixed.

We should also add an ignore directive to the files we don't want to format in the next uncrustify PR. It'd be much easier to do `uncrustify **/*.{c,h}` than to individually pick which files to format.

File: src/nvim/os/win_defs.h [hunk 1]
```diff
@@ -7,11 +7,11 @@
 
 // winsock2.h must be first to avoid incompatibilities
 // with winsock.h (included by windows.h)
```
Comment (justinmk) [inline src/nvim/os/win_defs.h:9]: note the comment , 

```
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\shared\ws2ipdef.h(127): warning C4005: 'IP_MULTICAST_TTL': macro redefinition [D:\a\neovim\neovim\build\src\nvim\nvim.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\um\winsock.h(344): note: see previous definition of 'IP_MULTICAST_TTL'
C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\shared\ws2ipdef.h(128): warning C4005: 'IP_MULTICAST_LOOP': macro redefinition [D:\a\neovim\neovim\build\src\nvim\nvim.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\um\winsock.h(345): note: see previous definition of 'IP_MULTICAST_LOOP'
```

**Review by justinmk — APPROVED** [review]
LGTM after win_defs.h is fixed

File: src/nvim/os/win_defs.h (hunk 1 above)
Comment (dundargoc) [reply src/nvim/os/win_defs.h:9]: :+1:

File: src/nvim/os/win_defs.h [hunk 2]
```diff
@@ -7,11 +7,14 @@
...
+
+// uncrustify:indent-off
 #include <winsock2.h>
-#include <windows.h>
-#include <sys/stat.h>
+// uncrustify:indent-on
```
Comment (justinmk) [inline src/nvim/os/win_defs.h:13]: this isn't indent-related. is `uncrustify:indent-off` the generic "disable uncrustify" flag? If so we should rename it to `uncrustify:off`

Comment (dundargoc) [reply src/nvim/os/win_defs.h:13]: It's like you read my mind :)

**Comment by justinmk** [comment]
waiting for GHA to run (windows especially)...

**Comment by dundargoc** [comment]
Something went wrong, it's not running. Can trigger the CI job manually?

**Comment by justinmk** [comment]
i looked around but don't see any way to run against a particular commit or PR. tried re-running a previous run.

**Comment by clason** [comment]
It might be because the (sole) commit is called `fixup`?

**Comment by dundargoc** [comment]
[GitHub Actions is down...](https://www.githubstatus.com/)

**Comment by clason** [comment]
Looks like GHA is b0rked? githubstatus.org lists them as "degraded"...

**Comment by justinmk** [comment]
reran ci

**Comment by dundargoc** [comment]
Let's get this merged if there's nothing else that needs fixing?

**Pull request merged**